src/core/public/http/http_service.test.ts

```diff
@@ -342,6 +346,7 @@ describe('interception', () => {
 
   it('should skip remaining interceptors when controller halts during request', async () => {
     const order: string[] = [];
+    let pending = true;
```

Comment:
Instead of tracking a flag, I find it more explicit if we passed mock functions into the the `then` and `catch` callbacks and assert that they don't get called.